routing/edge_estimator.hpp

```diff
@@ -21,17 +21,16 @@ namespace routing
 class EdgeEstimator
 {
 public:
-  explicit EdgeEstimator(double maxSpeedKMpH);
   virtual ~EdgeEstimator() = default;
 
-  double CalcHeuristic(m2::PointD const & from, m2::PointD const & to) const;
+  virtual double CalcHeuristic(m2::PointD const & from, m2::PointD const & to) const = 0;
```

Comment:
BTW, why do we need one more class? If you're going to override EdgeEstimator, just mark CalcHeuristic and CalcLeapWeight as virtual, and override them in your subclass. Moreover, please, mark current subclasses as final. Hope, this will somehow reduce the cost of virtual methods, because these methods are on the hot path in routing.

Reply:
+1